### PR TITLE
Update ORT model conversion script to support custom ops

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
Update ORT format model conversion script to support custom ops.
Simplify conversion now that we can process the ORT format file to determine the required ops.
Enable custom op registration via python in a minimal build with custom ops
Use pathlib in a few places when working with paths.

**Motivation and Context**
Better support for converting a model with custom ops.

### CODE REVIEW

File: tools/python/convert_onnx_models_to_ort.py [hunk 1]
```diff
@@ -128,17 +125,35 @@ def parse_args():
                         help='Add operator specific type information to the configuration file to potentially reduce '
                              'the types supported by individual operator implementations.')
 
-    parser.add_argument('model_path', help='Provide path to directory containing ONNX model/s to convert. '
-                                           'Files with .onnx extension will be processed.')
+    parser.add_argument('--custom_op_library', type=pathlib.Path, default=None,
```
Comment (hariharans29) [inline tools/python/convert_onnx_models_to_ort.py:128]: do we need to add a test for exercising this path ?

Comment (skottmckay) [reply tools/python/convert_onnx_models_to_ort.py:128]: Added

**Review by gwang-msft — APPROVED** [review]
:shipit:

**Comment by gwang-msft** [comment]
> The NNAPI Execution Provider is only able to execute ONNX operators using NNAPI. When creating the ORT format model it is recommended to limit the optimization level to 'basic' so that custom internal ONNX Runtime operators are not added by the 'extended' optimizations. This will ensure that the maximum number of nodes can be executed using NNAPI. See the [graph optimization](ONNX_Runtime_Graph_Optimizations.md) documentation for details on the optimization levels.

Since you are already touching this file, would you please update the dead link here, and the link in ONNX_Runtime_Mobile_NNAPI_perf_considerations.md as well

---
Refers to: docs/ONNX_Runtime_for_Mobile_Platforms.md:131 in d5b8f3a. [](commit_id = d5b8f3a45d1f0ced76113a606dfc4a9d3cff57dc, deletion_comment = False)

**Pull request merged**